### PR TITLE
Correctly encode percentage signs for Android and iOS

### LINKED ISSUE #30: Percentage encoding on Android
Hello,

Can percentages be added to the encoding substitution for the Android platform? 

Android Studio throws an error for strings that have a percentage in them but aren't format strings. An example:

`<string name="over_achiever">"Na het behalen van 100%"</string>`

If the `%` in the above string could become `&#37;` in the final strings XML file, that should fix the issue. And an exception should be added to not encode the percentage if it is included as part of a format string, as in `See all in %1{{s}}`

### PR DESCRIPTION
**Pull request by larslockefeer**
Fixes #30, #21 and #31

### CODE REVIEW

File: tests/actions/__snapshots__/render.test.js.snap [hunk 1]
```diff
@@ -8,7 +8,7 @@ Result {
 \\"Settings.PushPermissionsRequest.Subtitle.COPY\\" = \\"Stay up to date\\";
 \\"Something.With.An.Arbitrary.Amount.Of.Nesting.COPY\\" = \\"We support nesting as deep as you'd like\\";
 \\"Delivery.Widget.Title.COPY\\" = \\"Help: %1$@\\";
-\\"Delivery.Widget.SpecialCharacters.COPY\\" = \\"The following characters need special handling on some platforms: @ ? < & ' \\"\\";
+\\"Delivery.Widget.SpecialCharacters.COPY\\" = \\"The following characters need special handling on some platforms: @ ? < & ' \\". And then there is the percent-sign: most cases should be replaced (e.g. 99.9%% and %%1 and %%.1) but not the percentage-sign in strings like %1$@ or %2$d.\\";
```
Comment (larslockefeer) [inline tests/actions/__snapshots__/render.test.js.snap:11]: @johankool Here you can see how the `Localizable.strings` file will end up after this change. This is what you intended, right?

File: tests/actions/__snapshots__/render.test.js.snap [hunk 2]
```diff
@@ -38,7 +38,7 @@ It may also contain empty lines to break up text into paragraphs.\\";
   <string name=\\"Checkout_OrderOverview_Subtotal_COPY\\">\\"Total price before taxes: %1$s\\"</string>
   <string name=\\"Something_With_An_Arbitrary_Amount_Of_Nesting_COPY\\">\\"We support nesting as deep as you'd like\\"</string>
   <string name=\\"Delivery_Widget_Title_COPY\\">\\"Help: %1$s\\"</string>
-  <string name=\\"Delivery_Widget_SpecialCharacters_COPY\\">\\"The following characters need special handling on some platforms: @ ? &lt; &amp; ' &quot;\\"</string>
+  <string name=\\"Delivery_Widget_SpecialCharacters_COPY\\">\\"The following characters need special handling on some platforms: @ ? &lt; &amp; ' &quot;. And then there is the percent-sign: most cases should be replaced (e.g. 99.9&#37; and &#37;1 and &#37;.1) but not the percentage-sign in strings like %1$s or %2$d.\\"</string>
```
Comment (larslockefeer) [inline tests/actions/__snapshots__/render.test.js.snap:41]: @asvitzer: this is what it will do for Android.

**Comment by johankool** [comment]
@larslockefeer For iOS this works correct for us! Thanks! @maartenapple, can you check if this change also works correctly for Android with our files?

**Comment by maartenapple** [comment]
@johankool @larslockefeer it also works correctly for our Android files! 👍

**Comment by larslockefeer** [comment]
Cool! Thanks for the quick reply. I'll do a small follow-up commit, merge and cut a new release.

File: tests/actions/__snapshots__/render.test.js.snap (hunk 2 above)
Comment (asvitzer) [reply tests/actions/__snapshots__/render.test.js.snap]: Thanks, @larslockefeer! Works well :) Sorry for the delay.

**Pull request merged**